CMakeLists.txt

```diff
@@ -25,6 +25,8 @@ option(WITH_DEBUG_SYMBOLS "With debug symbols" ON)
 option(WITH_THRIFT "With thrift framed protocol supported" OFF)
 option(BUILD_UNIT_TESTS "Whether to build unit tests" OFF)
 option(DOWNLOAD_GTEST "Download and build a fresh copy of googletest. Requires Internet access." ON)
+option(BUILD_TOOLS "Whether to build tools" ON)
+option(WITH_SHARED_LIB "With dynamic lib" ON)
```

Comment:
Hi @x2c3z4 , Thanks for your contribution, could you please explain the reason for adding WITH_SHARED_LIB, thanks!

Reply:
Maybe it is better to be:
option(BUILD_SHARED_LIB "Whether to build shared lib" ON)

Reply:
> Hi @x2c3z4 , Thanks for your contribution, could you please explain the reason for adding WITH_SHARED_LIB, thanks!

I want to static link the brpc to my project, and my project has also static linked with leveldb/gflags/glog, so if I link the brpc shared lib, there are some errors that I should handle, so add a switch to control the shared lib is a good option.

Reply:
> Maybe it is better to be: option(BUILD_SHARED_LIB "Whether to build shared lib" ON)
OK.